smoke_zephyr/configuration.py

```diff
@@ -202,6 +202,63 @@ def save(self):
 		with open(self.configuration_file, 'w') as file_h:
 			file_h.write(self._serializer('dumps', self._storage))
 
+class MemoryConfiguration(Configuration):
+	"""
+	This class provides a generic object for parsing information from variables in multiple formats.
+	"""
+
+	def __init__(self, mem_object, object_type, prefix=''):
+		"""
+		:param str mem_object: The memory object to parse.
+		:param str prefix: String to be prefixed to all option names.
+		:param str object_type: String to identify type of serializer to use.
+		"""
+		self.prefix= prefix
+		self.seperator = '.'
+		self.object_type = object_type
+		if type(mem_object) != 'dict':
+			self._storage = dict(self._serializer('loads', mem_object))
+		else:
+			self._storage = mem_object
+
+	def _serializer(self, operation, *args):
+		if not self.object_type in SERIALIZER_DRIVERS:
+			raise ValueError('unknown type \'' + self.object_type + '\'')
+		function = SERIALIZER_DRIVERS[self.object_type][operation]
+		return function(*args)
+
+
+	def get_missing(self, mem_object, object_type):
+		"""
+		Use a verification configuration which has a list of required options
+		and their respective types. This information is used to identify missing
+		and incompatible options in the loaded configuration.
+
+		:param str mem_object: The file to load for verification data.
+		:param str object_type: Type to parse the mem_object as.
+		:return: A dictionary of missing and incompatible settings.
+		:rtype: dict
+		"""
+		vconf = MemoryConfiguration(mem_object, object_type)
+		missing = {}
+		for setting, setting_type in vconf.get('settings').items():
+			if not self.has_option(setting):
+				missing['missing'] = missing.get('settings', [])
+				missing['missing'].append(setting)
+			elif not type(self.get(setting)).__name__ == setting_type:
+				missing['incompatible'] = missing.get('incompatible', [])
+				missing['incompatible'].append((setting, setting_type))
+		return missing
+
+	def save(self, filename):
```

Comment:
There should not be a save method for `MemoryConfiguration`, persistence is what the `Configuration` class is for.